cmd/agent/app/flags.go

```diff
@@ -66,10 +67,17 @@ func (b *Builder) InitFromViper(v *viper.Viper) *Builder {
 		p.Workers = v.GetInt(prefix + suffixWorkers)
 		p.Server.QueueSize = v.GetInt(prefix + suffixServerQueueSize)
 		p.Server.MaxPacketSize = v.GetInt(prefix + suffixServerMaxPacketSize)
-		p.Server.HostPort = v.GetString(prefix + suffixServerHostPort)
+		p.Server.HostPort = toHostPort(v.GetString(prefix + suffixServerHostPort))
 		b.Processors = append(b.Processors, *p)
 	}
 
-	b.HTTPServer.HostPort = v.GetString(httpServerHostPort)
+	b.HTTPServer.HostPort = toHostPort(v.GetString(httpServerHostPort))
 	return b
 }
+
+func toHostPort(v string) string {
```

Comment:
What happens if someone is running on a IPV6 host with the address of the form `2001:0db8:85a3:0000:0000:8a2e:0370:7334` ?

Reply:
Perhaps we might check that the string starts with a `:` instead?

Reply:
so really I was trying to catch a raw number here, so I could instead try Long.parse()